plugins/BEdita/Core/src/Model/Table/ObjectsTable.php

```diff
@@ -170,4 +175,16 @@ public function findType(Query $query, array $options)
 
         return $query;
     }
+
+    /**
+     * Find by date range using `DateRanges` table findDate filter
+     *
+     * @param \Cake\ORM\Query $query Query object instance.
+     * @param array $options Array of acceptable date range conditions.
+     * @return \Cake\ORM\Query
+     */
+    public function findDateRanges(Query $query, array $options)
+    {
+        return TableRegistry::get('DateRanges')->findDateRanges($query, $options, $this->getAlias());
```

Comment:
Here I think it would probably be better something like:

```php
return $query->innerJoinWith('DateRanges', function (Query $query) use ($options) {
    return $query->find('dateRanges', $options);
});
```